test/unit/scan/test_folder_traversal.py

```diff
@@ -45,9 +48,82 @@ def test_flat_folder(self, tmp_path):
             fix_windows_path_limit(str(tmp_path / "dir" / "file3.txt")),
         ]
 
+    @pytest.mark.skipif(
+        platform.system() == 'Windows',
+        reason="Windows doesn't allow / or \\ in filenames",
+    )
+    def test_invalid_filename(self, tmp_path):
+
+        # Create a directory structure below with initial scanning point at tmp_path/dir:
+        # tmp_path
+        # └── dir
+        #     ├── file1.txt
+        #     ├── file2.txt
+        #     ├── file\bad.txt
+        #     └── file[DEL]bad.txt
+
+        (tmp_path / "dir").mkdir(parents=True)
+
+        (tmp_path / "dir" / "file1.txt").write_text("content1")
+        (tmp_path / "dir" / "file2.txt").write_text("content2")
+        (tmp_path / "dir" / "file\\bad.txt").write_text("bad1")
+        (tmp_path / "dir" / "file\x7fbad.txt").write_text("bad2")
+
+        reporter = ProgressReport(sys.stdout, False)
+        folder = LocalFolder(str(tmp_path / "dir"))
+        local_paths = folder.all_files(reporter=reporter)
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert reporter.has_warnings_or_errors()
+        assert len(reporter.warnings) == 2
+        assert "has an invalid filename (file names must not contain '\\'). Skipping" in reporter.warnings[
```

Comment:
please assert that not only explanation what is wrong to the user, but also filename with path is shown to them so they can go to the exact file and fix it